### DISCUSSION TITLE
系统监控是否能增加进程的占用

### DISCUSSION THREAD

**Discussion by seth-shi** [1Panel-dev/1Panel#6887] [Ideas]
### 1Panel 版本

v1.10.20-lts

### 请描述您的需求或者改进建议

监控有时候死机了重启, 想排查是哪个进程占用, 但是现在只能看到 CPU 占用,无法查看具体是哪个进程

**Comment by wanghe-fit2cloud** [comment]
上述需求是希望在 1Panel 系统中增加查看历史进程占用的功能吗？

**Comment by seth-shi** [comment]
嗯嗯
可以有一个几秒一次快照

**Comment by wanghe-fit2cloud** [comment]
> 嗯嗯 可以有一个几秒一次快照

我们先在本地调研一下，看看是否有可行的方案。

**Comment by seth-shi** [comment]
好的, 感谢

**Comment by wanghe-fit2cloud** [comment]
实现上述需求可能较为复杂，以下是 ChatGPT 给出的参考方案，可以先看看是否可行：

---

`sysstat` 默认不记录每个进程的具体资源使用情况。要监控和查看历史进程信息，可以结合以下工具和步骤：

1. **安装 `sysstat`**：
   这是用来记录系统整体的 CPU、内存、I/O 等历史性能数据。
   
2. **安装 `psacct`（Process Accounting）**：
   `psacct`（或 `acct`）工具能够记录每个进程的使用情况，比如 CPU 时间、内存等。安装后会自动记录进程信息，使用以下命令安装：

   ```bash
   sudo apt install acct     # 在 Ubuntu/Debian
   sudo yum install psacct   # 在 CentOS/RHEL
   ```

3. **启用 `psacct`**：
   启用该服务后，它会自动记录每个进程的历史数据：

   ```bash
   sudo systemctl start psacct
   sudo systemctl enable psacct
   ```

4. **查看进程历史**：
   使用以下命令可以查看每个进程的历史资源使用：

   - **查看命令执行历史**：
     ```bash
     lastcomm
     ```
   
   - **查看用户级的 CPU 和内存使用统计**：
     ```bash
     sa
     ```

   - **查看各进程的详细信息**：
     ```bash
     ac -p           # 按用户查看 CPU 时间
     ac -d           # 按天查看 CPU 时间
     ```

`psacct` 提供的这些记录能帮助分析进程的历史数据。不过，由于 `sysstat` 默认不监控进程，结合 `psacct` 才能更详细地跟踪到具体进程的历史情况。

**Reply by seth-shi** [reply]
不好用, 就像我 top 也能看进程占用, 装 1panel 就是为了方便的

**Reply by wanghe-fit2cloud** [reply]
后续再研究一下吧，目前看来可能没有太好的方案用来来查看历史进程占用情况。

**Reply by seth-shi** [reply]
宝塔开源有这个功能，可以借鉴一下，感觉就 top n 秒正则匹配记录一下

**Reply by wanghe-fit2cloud** [reply]
好的，能否提供一下功能截图？

**Comment by seth-shi** [comment]
![image](https://github.com/user-attachments/assets/86969473-8315-4987-85a9-478a0206c21d)
https://www.bt.cn/new/product_linux.html

**Reply by seth-shi** [reply]
@wanghe-fit2cloud
